android/src/main/java/com/swmansion/rnscreens/ScreenFragment.java

```diff
@@ -66,6 +66,47 @@ public Screen getScreen() {
     return mScreenView;
   }
 
+  public void onContainerUpdate() {
+    ScreenStackHeaderConfig config = findHeaderConfig();
+    boolean configInChild = checkIfChildWithConfig(getScreen());
+    if (!configInChild && config != null && config.getScreenFragment().getActivity() != null) {
+      // if there is no child with config, we look for a parent with config to set the orientation
+      config.getScreenFragment().getActivity().setRequestedOrientation(config.getScreenOrientation());
+    }
+  }
+
+  private ScreenStackHeaderConfig findHeaderConfig() {
+    ViewParent parent = getScreen().getContainer();
+    while (parent != null) {
+      if (parent instanceof Screen) {
+        for (int i = 0; i < ((Screen) parent).getChildCount(); i++) {
+          View child = ((Screen) parent).getChildAt(i);
+          if (child instanceof ScreenStackHeaderConfig) {
+            return (ScreenStackHeaderConfig) child;
+          }
+        }
+      }
+      parent = parent.getParent();
+    }
+    return null;
+  }
+
+  protected static boolean checkIfChildWithConfig(View view) {
```

Comment:
rename this to make it more descriptive. Do we really want to traverse all view hierarchy? Maybe it should be enought to only traverse fragments hierarchy?